core/FrontController.php

```diff
@@ -189,29 +189,10 @@ public static function triggerSafeModeWhenError()
     /**
      * Loads the config file
      * This is overridden in tests to ensure test config file is used
-     *
-     * @return Exception
      */
     public static function createConfigObject()
     {
-        $exceptionToThrow = false;
-        try {
-            Config::getInstance()->database; // access property to check if the local file exists
-        } catch (Exception $exception) {
-            Log::debug($exception);
-
-            /**
-             * Triggered when the configuration file cannot be found or read, which usually
-             * means Piwik is not installed yet.
-             *
-             * This event can be used to start the installation process or to display a custom error message.
-             *
-             * @param Exception $exception The exception that was thrown by `Config::getInstance()`.
-             */
-            Piwik::postEvent('Config.NoConfigurationFile', array($exception), $pending = true);
-            $exceptionToThrow = $exception;
-        }
-        return $exceptionToThrow;
+        Config::getInstance()->database; // access property
```

Comment:
All environment validation should be done by EnvironmentValidator; I was just afraid of changing yet another thing and risking side effects. I guess we shouldn't need anything really, but could keep Config::getInstance() for safety (we'll be able to remove when it's put in DI).